### DISCUSSION TITLE
[Help Wanted] Debian package request

### DISCUSSION THREAD

**Discussion by port19x** [qtile/qtile#3358] [Ideas]
It would be great to have a debian package for qtile again.
This would drastically ease the installation and lead to higher adoption in the long run.
It's important that the package is maintained and doesn't get abandoned like the [previous](https://tracker.debian.org/pkg/qtile) [two](https://launchpad.net/~tycho-s/+archive/ubuntu/ppa) attempts.

Some good entrypoints to relevant docs:
[debian new maintainers guide](https://www.debian.org/doc/manuals/maint-guide/)
[debian maintainers guide](https://www.debian.org/doc/manuals/debmake-doc/index.en.html)
[debian policy](https://www.debian.org/doc/debian-policy/)
[debian wiki packaging](https://wiki.debian.org/Packaging)

Existing packages:
[arch aur](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=qtile-git) (qtile 0.20.0)
[opensuse](https://build.opensuse.org/package/show/openSUSE%3AFactory/qtile) (qtile 0.18.1)
[freebsd](https://cgit.freebsd.org/ports/tree/x11-wm/qtile) (qtile 0.18.1)
[fedora](https://src.fedoraproject.org/rpms/qtile) (qtile 0.14.2)

_A maintained rpm package also wouldn't hurt_

**Comment by elParaguayo** [comment]
You need to pick this up with the debian team. We're not responsible for packaging for the distros here.

But I do agree, I'd like more distros to have this in their repos!

**Reply by port19x** [reply]
> You need to pick this up with the debian team. We're not responsible for packaging for the distros here.
> 
> But I do agree, I'd like more distros to have this in their repos!

Actually I would like to disagree.
In the real world, no distro maintainer goes out of his way to create packages for cool software that exists.
Meaning if you want your software in repos you are in fact responsible to get it packaged. And even then distros can be quite picky and may have high standards of what gets accepted.

I still believe that you and mcol better spend your time on general maintenance, but believing the debian team would even react to "hey, please package my software for me" is kind of ridicolous and not reflective of how distros operate

**Reply by elParaguayo** [reply]
Not what I meant.

The packaging would not be in this repo and so we would not do the packaging ourselves.

I'm also not a Debian user and don't know how to package for Debian, nor am I particularly interested in learning how to do it.

If a Debian user wants it in their repo then my hope would be that they develop the package and submit it to the maintainers of the repos. 

I would be very happy to address any queries that user had to help support them (and we have provided to support to the Arch packagers).

We will also happily include instructions in our docs. 

My only point was that I would not expect to develop/maintain/host any distro-specific packaging files in this repo.

This is a discussion forum so happy for people to disagree!

**Reply by port19x** [reply]
Oh okay, I misunderstood you then and we are on the same page after all. Not hosting distro specific packaging files here makes sense now that I think about it

**Reply by port19x** [reply]
Also technically Solus operates like that, but their repositories are severly lacking as a result

**Reply by elParaguayo** [reply]
No problem. My original answer was far too short to really explain what I meant!

**Comment by port19x** [comment]
Yeah, thats not something you or mcol should ~~waste~~ spend your time on.

I can do some groundwork on this, but want to invite anyone looking for a way to contribute to leave a comment and contact me

> You need to pick this up with the debian team. We're not responsible for packaging for the distros here.
> 
> But I do agree, I'd like more distros to have this in their repos!

**Comment by port19x** [comment]
Sure, let's have it as a discussion first

**Comment by epsilon77** [comment]
Hello Team!

I recently discovered qtile, I started using it and I would like include it in Debian again, for that, I was working in a first version[1] without wayland support because some depends aren't in debian yet.  
 
For wayland support, I include some depends[2] [3], but we still need pywlroots(is already in NEW Queue)[4]

What are the next steps?

The freeze is coming, for, I planning upload the initial version (without wayland support)  to experimental soon, and once the pywlroot will be accepted in unstable, I will try prepare a new version to both backends.

I hope can include qtile for the next debian release, but we depend on some timing, so we just hope we have a little luck.

For now all feedback are welcome.
 
Thanks for work in qtile!
 
 Regards.
 
[1]: https://salsa.debian.org/epsilon/qtile
[2]: https://tracker.debian.org/pkg/python-xkbcommon
[3]: https://tracker.debian.org/pkg/pywayland
[4]: https://ftp-master.debian.org/new/pywlroots_0.17.0-1.html

**Reply by elParaguayo** [reply]
Hi @epsilon77 

Thanks for the message. It would be great to have qtile in the Debian repo so we appreciate you looking into this.

Copying @jwijenbergh as he's better on the Wayland stuff than me but, in short, you need:
* wlroots 0.17
* pywayland
* pywlroots 0.17
* python-xkbcommon

qtile's wayland backend then needs to be built by running `scripts/ffi_build.py`.

We also have some `.desktop` files to help users with starting qtile from their display manager.

Let us know if you run into build/test errors and we'll try to help you resolve them.

**Comment by epsilon77** [comment]
Hello!

Recently, I uploaded qtile 0.30 to Debian experimental, and it was accepted by the ftp-master team[1]. I would like someone to help test this version a bit to ensure that it works correctly before we move it to unstable. You can find information on how to test packages from experimental at this link [2].

For now, it doesn't have Wayland support because, unfortunately, we still don't have pywlroots in Debian (this is something to discuss since wlroots version 0.17 will soon be removed from Debian), but it will be useful for all X11 users.

All feedback are welcome.

Regards!

[1]: https://tracker.debian.org/news/1619885/accepted-qtile-0300-1-source-all-into-experimental/
[2]: https://wiki.debian.org/DebianExperimental#To_install_an_experimental_package
